etna/search/forms.py

```diff
@@ -198,20 +237,238 @@ class BaseCollectionSearchForm(forms.Form):
         required=False,
     )
 
+    def _get_compressed_date_or_none_on_error(
+        self, start_day: str, start_month: str, start_year: str
+    ) -> Union[None, datetime]:
+        """
+        Assumption: all params not required to belong to a real date. The Exception is bypassed to return None
+        as error messages are customised for date and handled outside this method.
+        Returns: valid date if all fields belong to a real date, otherwise None if invalid date.
+        """
+        try:
+            return datetime(
+                day=int(start_day), month=int(start_month), year=int(start_year)
+            )
+        except Exception as e:
+            logger.debug(
+                f"Error bypassed in _get_compressed_date_or_none_on_error: start_day={start_day}, start_month={start_month}, start_year={start_year} exception={e}"
+            )
+            return None
+
+    def get_cleaned_date_parts(
+        self, prefix_field_with_type: str
+    ) -> tuple[str, str, str]:
+        """
+        Returns tuple of cleaned seperated date fields
+
+        Args:
+
+        prefix_field_with_type:
+        example when field is 'opening_start_day', prefix_field is 'opening', prefix_field_with_type is 'opening_start'
+        """
+        return (
+            self.cleaned_data.get(prefix_field_with_type + "_day"),
+            self.cleaned_data.get(prefix_field_with_type + "_month"),
+            self.cleaned_data.get(prefix_field_with_type + "_year"),
+        )
```

Comment:
This feels a little bit like code smell, as there's no way to enforce this naming convention for fields. Is it not possible to have this wrapped up in a single form field, where this behaviour can be controlled (as in the examples provided). Surely we could just copy this code exactly and tweak the validation logic to meet our needs?:
Field: https://github.com/wildfish/crispy-forms-gds/blob/master/src/crispy_forms_gds/fields.py#L11-L170
Widget: https://github.com/wildfish/crispy-forms-gds/blob/master/src/crispy_forms_gds/widgets.py#L5-L59